File: backend/app/schemas/mcp_tools.py
from __future__ import annotations

from pydantic import BaseModel, Field


class McpToolConfigRequest(BaseModel):
    name: str = Field(..., min_length=1)
    description: str | None = None
    toolType: str | None = "custom"
    toolName: str = Field(..., min_length=1)
    transport: str | None = "stdio"
    command: str = Field(..., min_length=1)
    args: list[str] | None = None
    env: dict[str, str] | None = None
    enabled: int | None = 1


class McpToolConfigResponse(BaseModel):
    id: int
    name: str
    description: str | None = None
    toolType: str | None = "custom"
    toolName: str
    transport: str | None = "stdio"
    command: str
    args: list[str] | None = None
    env: dict[str, str] | None = None
    enabled: int | None = 1
    preset: int | None = 0
    createdAt: str | None = None
    updatedAt: str | None = None


class AgentPlanWebSearchMcpRequest(BaseModel):
    name: str | None = None
    description: str | None = None
    apiKey: str = Field(..., min_length=1)


class McpToolTestRequest(BaseModel):
    query: str | None = None

File: frontend/src/api/mcpTools.ts
import api from '../utils/request';

export interface McpToolConfig {
  id: number;
  name: string;
  description?: string;
  toolType: string;
  toolName: string;
  transport: string;
  command: string;
  args: string[];
  env: Record<string, string>;
  enabled: number;
  preset: number;
  createdAt?: string;
  updatedAt?: string;
}

interface ApiResult<T> {
  code: number;
  message: string;
  data: T;
}

export const getMcpTools = (): Promise<ApiResult<McpToolConfig[]>> =>
  api.get('/api/v1/mcp-tools');

export const createAgentPlanWebSearchMcp = (data: {
  name?: string;
  description?: string;
  apiKey: string;
}): Promise<ApiResult<McpToolConfig>> =>
  api.post('/api/v1/mcp-tools', {
    name: data.name || 'Agent Plan Web Search',
    description: data.description || '',
    toolType: 'agent_plan_web_search',
    toolName: 'web_search',
    transport: 'stdio',
    command: 'uvx',
    args: ['git+https://github.com/volcengine/mcp-server#subdirectory=server/mcp_server_askecho_search_infinity'],
    env: { API_KEY: data.apiKey },
    enabled: 1,
  });

export const deleteMcpTool = (id: number): Promise<ApiResult<void>> =>
  api.delete(`/api/v1/mcp-tools/${id}`);

export const testMcpTool = (
  id: number,
  data: { query?: string }
): Promise<ApiResult<Record<string, unknown>>> =>
  api.post(`/api/v1/mcp-tools/${id}/actions/test`, data);
